### ISSUE TITLE
Parcel imports the wrong package

### ISSUE THREAD

**Issue by domoritz** [parcel-bundler/parcel#1596]
# 🐛 bug report

It's hard to say exactly why this is happening but parcel imports are often not the right package and that causes errors in my app. Unfortunately, my code is not open source yet. I'll keep adding to this issue as I encounter more of these problems. The way I resolve them right now is to delete the cache and start parcel again. 

## 😯 Current Behavior

Here is one example. `/guide-mark` should be another javascript file but for some reason, it is an arrow file that I load data from. 

<img width="936" alt="screen shot 2018-06-21 at 17 39 43" src="https://user-images.githubusercontent.com/589034/41752241-0beb0592-757b-11e8-8cb8-f0265b55fad8.png">

TODO @domoritz, add more examples

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.9.0 / 1.9.2
| Node             | 10
| Operating System | MacOS

**Comment by emilniklas** [comment]
I'm not sure if this is related, but I'm stuggling with a problem where Parcel generates the same ID multiple times in the output. Basically the generated code looks like this:

```javascript
parcelRequire = function(...) {
  ...
}({
  1234: ...,
  1672: ...,
  1583: ...,
  1234: ...
})
```

As you can see, the object sent into the `parcelRequire` function has two keys called `1234`. JS allows this, but only actually includes the last occuring key/value. So if the module referred to as `1672` here includes a generated `require(1234)`, referring to the first occurrence, then the last occurrence will be returned from the `require`.

I definitely saw the type of issues that you're describing when it started happening for me. What's worse is that results vary between compilations, indicating some type of race condition. However, setting `PARCEL_WORKERS=1` doesn't help either.

I think I've narrowed it down to having to do with https://github.com/parcel-bundler/parcel/blob/master/src/Asset.js#L23 but I don't see why setting `PARCEL_WORKERS=1` wouldn't work in that case.

**Comment by emilniklas** [comment]
I seem to have "solved" it by using `--no-cache`, but I might've just gotten lucky with the race condition. Makes sense, though, if `Asset` instances are cached as-is.

**Comment by devongovett** [comment]
dup of #1656. should be fixed by #1694.
